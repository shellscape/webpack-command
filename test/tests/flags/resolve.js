const { apply, build, crcDist, test, validate } = require('../../util');

test('--resolve-*', module, () => {
  for (const name of ['resolve-alias', 'resolve-extensions', 'resolve-loader-alias']) {
    const fixture = `resolve/${name}`;
    const opts = { fixture };

    let config;

    it(`${name} reporter should validate`, () => {
      expect(validate(opts)).toEqual(true);
    });

    it(`${name} reporter should apply`, () => {
      config = apply(opts);

      expect(config).toMatchSnapshot();
    });

    it(`${name} reporter should build`, () =>
      build(config).then((result) => {
        expect(result).toMatchSnapshot();
        expect(crcDist()).toMatchSnapshot();
      }));
  }
});
