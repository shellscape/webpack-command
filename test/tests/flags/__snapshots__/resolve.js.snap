// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --resolve-* > resolve-alias reporter should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/resolve/entry.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
  "resolve": Object {
    "alias": Object {
      "aliased": "<PROJECT_ROOT>/test/fixtures/flags/resolve/aliased",
    },
  },
}
`;

exports[`Flags > --resolve-* > resolve-alias reporter should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.61 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/flags/resolve/aliased/dependency.js] 39 bytes {main} [built]
[./test/fixtures/flags/resolve/entry.js] 150 bytes {main} [built]"
`;

exports[`Flags > --resolve-* > resolve-alias reporter should build #1 1`] = `"23f35c9e"`;

exports[`Flags > --resolve-* > resolve-extensions reporter should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/resolve/entry-extensions.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
  "resolve": Object {
    "extensions": Array [
      ".custom",
    ],
  },
}
`;

exports[`Flags > --resolve-* > resolve-extensions reporter should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.64 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/flags/resolve/dependency.custom] 38 bytes {main} [built]
[./test/fixtures/flags/resolve/entry-extensions.js] 144 bytes {main} [built]"
`;

exports[`Flags > --resolve-* > resolve-extensions reporter should build #1 1`] = `"4ae3815"`;

exports[`Flags > --resolve-* > resolve-loader-alias reporter should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/resolve/entry-loader-alias.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
  "resolveLoader": undefined,
}
`;

exports[`Flags > --resolve-* > resolve-loader-alias reporter should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  20.5 KiB    main  [emitted]  main
Entrypoint main = main.js
[./node_modules/path-browserify/index.js] 8.54 KiB {main} [built]
[./node_modules/process/browser.js] 5.29 KiB {main} [built]
[./test/fixtures/flags/resolve sync recursive ^aliased!.*$] ./test/fixtures/flags/resolve sync ^aliased!.*$ 160 bytes {main} [built]
[./test/fixtures/flags/resolve/entry-loader-alias.js] 283 bytes {main} [built]"
`;

exports[`Flags > --resolve-* > resolve-loader-alias reporter should build #1 1`] = `"6c50b3b1"`;
